app/src/services/biconomy.ts

```diff
@@ -0,0 +1,124 @@
+import axios from 'axios'
+
+interface CreateProxyAndExecParams {
+  data: string
+  fallbackHandlerAddress: string
+  from: string
+  masterCopyAddress: string
+  operation: string
+  predeterminedSaltNonce: string
+  proxyFactoryAddress: string
+  signature: string
+  to: string
+  value: string
+}
+
+interface ExecTransactionParams {
+  data: string
+  dataGas: number
+  from: string
+  gasPrice: number
+  gasToken: string
+  operation: string
+  proxyAddress: string
+  proxyFactoryAddress: string
+  refundReceiver: string
+  safeTxGas: number
+  signature: string
+  to: string
+  value: string
+}
+
+class BiconomyService {
+  gasLimit = '3000000'
+
+  request = async (payload: any) => {
+    const BICONOMY_API_KEY = 'MPEEl225y.19edc8a0-06f9-444b-b6bc-c1b7ca6bdc31'
```

Comment:
Does having this visible leave us vulnerable to a DoS attack?

Reply:
Yeah definitely, it looks like the only preventative are these daily limits that can be set in the Biconomy dashboard:

<img width="1390" alt="Screen Shot 2021-03-03 at 10 31 52 am" src="https://user-images.githubusercontent.com/72715177/109729243-b1899380-7c0b-11eb-954b-50f91dc67580.png">

it looks like we can whitelist the proxy factory contract, but even then someone could spam the proxy factory... 🤔

Reply:
So I'm unsure if the user limit is per address or per IP address, can't find much info about it. Even per IP address an attacker can use rotating proxies I guess.

hmmmm soooo yeah potential attack - spam until dapp limit is depleted and no one can use the app. Good catch @KadenZipfel 

Will think about how we can obfuscate the api key.. might have deploy a tiny api between omen and biconomy.

Reply:
why can't we put the API key in a process.env file and then add it to .gitignore?

Reply:
good idea @gem1no but this would still expose the api key to an attacker, they could snoop it from the network tab in chrome tools when requests are made for example.

Reply:
Per user API limit is on user address not IP address and apart from Limit module, you can add the domain restrictions as well..
[https://docs.biconomy.io/guides/biconomy-dashboard#domain-restrictions](https://docs.biconomy.io/guides/biconomy-dashboard#domain-restrictions)

So only requests from your domain will be entertained for your API Key but still there are chances of someone impersonating the domain origin of the request.

As @hexyls  suggested, best way would be to deploy a tiny api between omen and biconomy to completely obfuscate api key.